h2o-core/src/main/java/water/server/ServletMeta.java

```diff
@@ -9,10 +9,21 @@
 
   private final String _contextPath;
   private final Class<? extends HttpServlet> _servletClass;
+  private final boolean _alwaysEnabled;
 
-  public ServletMeta(String contextPath, Class<? extends HttpServlet> servletClass) {
+  /**
```

Comment:
A constructor with default value false for alwaysEnabled would make sense since the default is usually what you want.

Reply:
Two constructors already on three parameters. I originally had it like that and removed it. It feels awkward to propose builder pattern anywhere. I made it so in https://github.com/h2oai/h2o-3/pull/5134/commits/8668ae733cabc6cd257c1c1946903d503d85722b .

Reply:
I think changes to this class are not a big problem, however, we need to keep in mind it is an extension point and might be used by 3rd parties.

Keeping the simplest constructor makes sense to me. More complex use can have a builder pattern for example.

Reply:
Question is whether to introduce it now or not. Again, there will be many opinions. I'd stick with the current two-constructor solution and then introduce a builder pattern. Some might even say there is either the constructor approach **or** builder with no constructors .... often I simply don't know what is wrong.

I'd say currently we're fine. One more and there will be builder together with constructors. Is that something acceptable ?

Reply:
IMHO feel free to introduce now. As long as the original constructor that would be used in 99.9% scenarios stays - I am fine with anything ;)

Reply:
I think we're fine now.

Reply:
Added one now.